### DISCUSSION TITLE
Add new sensor device class: Moisture

### DISCUSSION THREAD

**Discussion by Ernst79** [home-assistant/architecture#793] [General] [answered]
I'm adding support for moisture sensors for the new BThome integration and we had a discussion on discord about the need for a device class "moisture". I would like to propose to add a device class "moisture" with "%" as unit of measurement. A quick search on github learned me that, the following integrations use a moisture sensor

- plant
- sia
- rachio
- onewire
- wirelesstag
- zha
- arwn
- mysensors
- bthome
- xiaomi_ble

**Comment by MartinHjelmare** [comment, marked as answer]
I agree. 👍

**Reply by Jc2k** [reply]
Yep me too 👍

**Reply by balloob** [reply]
Same 👍

**Comment by MartinHjelmare** [comment]
Please look at some old merged core PRs that added a sensor device class and then go ahead and open PRs in core, frontend, dev docs and user docs.

**Reply by Ernst79** [reply]
Ok, will do that, thanks.

**Comment by sgbirch** [comment]
Is moisture a synonym of humidity?

**Reply by Ernst79** [reply]
No, apparently there is a difference. I think this site explains it nicely. 

https://basementissues.com/moisture-vs-humidity/

Moisture =  the presence of **liquid** (commonly water) in an environment (wood/soil/etc).
Humidity = the amount of water **vapor** that is present in the air

**Reply by emontnemery** [reply]
@Ernst79 Please update the description to be more precise about what the proposed "moisture" sensor represents, is it reporting the ratio of water in %?

**Reply by Ernst79** [reply]
The definition according to the Cambridge dictionary is 

> a [liquid](https://dictionary.cambridge.org/dictionary/english/liquid) such as [water](https://dictionary.cambridge.org/dictionary/english/water) in the [form](https://dictionary.cambridge.org/dictionary/english/form) of very [small](https://dictionary.cambridge.org/dictionary/english/small) [drops](https://dictionary.cambridge.org/dictionary/english/drop), either in the [air](https://dictionary.cambridge.org/dictionary/english/air), in a [substance](https://dictionary.cambridge.org/dictionary/english/substance), or on a [surface](https://dictionary.cambridge.org/dictionary/english/surface):

Or in the Merrian Webster dictionary

> liquid diffused or condensed in relatively small quantity

I think, in HA, the most common use will be a soil moisture sensor.

So, I propose to use:
MOISTURE: % of water in a substance 

For comparison, the current definition of humidity in HA is:
HUMIDITY: % of humidity in the air

Feel free to do a better proposal, as language wasn't my best subject in school.
